test/e2e/util.go

```diff
@@ -350,3 +426,32 @@ func findAWSWebACLRecursive(wafClient *wafv2.Client, aclName string, nextMarker
 	}
 	return nil, nil
 }
+
+// albcCredentialsRequest returns CredentialsRequest with IAM policies required by ALBC.
+func albcCredentialsRequest(name types.NamespacedName, secretRef corev1.ObjectReference, saName string) (*cco.CredentialsRequest, error) {
+	credentialsRequest := &cco.CredentialsRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+		},
+		Spec: cco.CredentialsRequestSpec{
+			ServiceAccountNames: []string{saName},
+			SecretRef:           secretRef,
+		},
+	}
+
+	codec, err := cco.NewCodec()
+	if err != nil {
+		return nil, err
+	}
+
+	providerSpec, err := codec.EncodeProviderSpec(&cco.AWSProviderSpec{
+		StatementEntries: albc.GetIAMPolicy().Statement,
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I decided to stay consistent with the rest of the e2e test, it doesn't panic, it fails individual tests letting the engineer to discover all the problems. The fail fast approach (`panic`) is indeed another way to go. Maybe it'll make more sense in the future when the e2e will be too long to wait, not it still takes reasonable mount of time.

Reply:
This is different from your usual E2E test failure though; the codec and provider spec are independent of the test environment, and they're effectively static data, so an error from `cco.NewCodec` or from `codec.EncodeProviderSpec` can only happen if you have a bug in the test logic itself.  It's akin to having a regular expression and using `regexp.MustCompile` so that any bugs in your regular expression cause your program to fail immediately (in other languages, it would even be a compile-time error; maybe one day Go will adopt Zig's `comptime` keyword).  To illustrate my point, one could even refactor `albcCredentialsRequest` to make the nature of these errors more apparent:
```go
var defaultProviderSpec = func() *runtime.RawExtension {
	codec, err := cco.NewCodec()
	if err != nil {
		panic(err)
	}

	providerSpec, err := codec.EncodeProviderSpec(&cco.AWSProviderSpec{
		StatementEntries: albc.GetIAMPolicy().Statement,
	})
	if err != nil {
		panic(err)
	}

	return providerSpec
}()

// albcCredentialsRequest returns CredentialsRequest with IAM policies required by ALBC.
func albcCredentialsRequest(name types.NamespacedName, secretRef corev1.ObjectReference, saName string) (*cco.CredentialsRequest, error) {
	credentialsRequest := &cco.CredentialsRequest{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name.Name,
			Namespace: name.Namespace,
		},
		Spec: cco.CredentialsRequestSpec{
			ServiceAccountNames: []string{saName},
			SecretRef:           secretRef,
		},
	}

	credentialsRequest.Spec.ProviderSpec = defaultProviderSpec

	return credentialsRequest, nil
}
```
It's fine if you still prefer to leave `albcCredentialsRequest` as is, but I wanted to clarify that I view these errors as belonging to a different class of errors vis-à-vis errors related to E2E test failures.

Reply:
Added `panic` and renamed the function to `mustGenerateALBCCredentialsRequest`.